aineko/extras/connectors/http_poller.py

```diff
@@ -0,0 +1,257 @@
+            # so that it can be properly caught by the node
+            # Note: this is required because pydantic errors
+            # are not pickleable
+            raise ValueError(
+                "Failed to cast params to ParamsHTTPPoller type. "
+                f"The following error occurred: {err}"
+            ) from err
+
+        # Ensure only one output dataset is provided
+        output_datasets = [
+            dataset for dataset in self.producers.keys() if dataset != "logging"
+        ]
+        if len(output_datasets) > 1:
+            raise ValueError(
+                "Only one output dataset is allowed for the "
+                "HTTPoller connector. "
+                f"{len(output_datasets)} datasets given."
+            )
+        self.output_dataset = output_datasets[0]
+
+        # Create a session
+        self.log(
+            f"Creating new session to endpoint {self.http_poller_params.url}."
+        )
+        self.session = requests.Session()
+
+    def _execute(self, params: Optional[Dict] = None) -> None:
+        """Polls and gets data from the HTTP or HTTPS endpoint.
+
+        Raises:
+            Exception: If the retry count exceeds the max retries.
+        """
+        # Check if it is time to poll
+        if (
+            time.time() - self.last_poll_time
+            >= self.http_poller_params.poll_interval
+        ):
+            # Update the last poll time
+            self.last_poll_time = time.time()
+
+            try:
+                # Poll HTTP endpoint
+                response = self.session.get(
+                    self.http_poller_params.url,
+                    timeout=self.http_poller_params.timeout,
+                    headers=self.http_poller_params.headers,
+                    data=self.http_poller_params.data,
+                    params=self.http_poller_params.params,
+                    json=self.http_poller_params.json_,
+                    auth=self.http_poller_params.auth,
+                )
+                # Check if the request was successful
+                if (
+                    response.status_code
+                    not in self.http_poller_params.success_codes
+                ):
+                    # pylint: disable=broad-exception-raised
+                    raise Exception(
+                        f"Request to url {self.http_poller_params.url} "
+                        "failed with status code: "
+                        f"{response.status_code}"
+                    )
+                raw_message = response.text
+            except Exception as err:  # pylint: disable=broad-except
+                # If request fails, log the error and sleep
+                self.log(
+                    "Request failed. "
+                    f"Sleeping for {self.http_poller_params.poll_interval} "
+                    f"seconds. Error: {err}",
+                    level="error",
+                )
+                time.sleep(self.http_poller_params.poll_interval)
```

Comment:
It's not clear to me why we use `poll_interval` here but `retry_sleep` in the exception handling below. 

If we're going to handle the cases differently then the documentation for `retry_sleep` should specify that it is only for parsing / writing to datasets.

In my mind, `The number of seconds to wait between retries.` implies that `retry_sleep` would be used whenever we need to retry something.

Reply:
Also, if an endpoint is rate limiting the polls we might want to give the user more flexibility to wait longer than in the general case (and this would be tunable via the separate `retry_sleep` param)

Reply:
this was a mistake, i replaced poll_interval with retry_sleep here. im going to follow up with exponential backoff which should help in the rate limiting scenario you desribe.